### ISSUE TITLE
PyTorch 1.4 support

### ISSUE THREAD

**Issue by trobro** [CharlesShang/DCNv2#45]
I got the same error when compiling as in #44 . Maybe because I'm using PyTorch 1.4? Anayways, I have a fix for it, will create a PR.

My system:

Windows 7
Visual Studio 2017
Anaconda Python 3.7
Cuda 10.0
PyTorch 1.4

**Comment by IgorDavidyuk** [comment]
Hello, @trobro !
Thanks for your fix!
Allow me to ask for your advice, please.
Situation description: I know that deformable convolution is supported natively in Pytorch since 1.4, but this does not mean DCN may be extracted to a static graph via jit. Moreover,  DCN added to Pytorch is version 1, not this one (version two).
Question: Can you confirm I understand the situation correctly and how hard is to add DCNv2 support to torchscript from your point of view?

**Comment by IgorDavidyuk** [comment]
I will also create an issue with the same question.

**Comment by trobro** [comment]
Glad I could help @IgorDavidyuk ! But when it comes to torchscript I myself have no knowledge unfortunately.

**Comment by jinfagang** [comment]
```
DCNv2/src/cuda/dcn_v2_cuda.cu(107): error: identifier "THCState_getCurrentStream" is undefined

```

Any updates on this issue?

**Comment by aligoglos** [comment]
I got below error with this environment : 
```
Windows 7
Visual Studio 2019
Anaconda Python 3.7
Cuda 10.1
PyTorch 1.4
```

```
running install
running bdist_egg
running egg_info
writing DCNv2.egg-info\PKG-INFO
writing dependency_links to DCNv2.egg-info\dependency_links.txt
writing top-level names to DCNv2.egg-info\top_level.txt
reading manifest file 'DCNv2.egg-info\SOURCES.txt'
writing manifest file 'DCNv2.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
running install_lib
running build_ext
building '_ext' extension
Emitting ninja build file D:\Artificial Intelligence\SuperResolution\Video\TDAN\src\DCNv2\build\temp.win-amd64-3.7\Release\build.ninja...
Compiling objects...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: error: build.ninja:21: multiple outputs aren't (yet?) supported by depslog; bring this up on the mailing list if it affects you

Traceback (most recent call last):
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\utils\cpp_extension.py", line 1400, in _run_ninja_build
    check=True)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 71, in <module>
    cmdclass={"build_ext": torch.utils.cpp_extension.BuildExtension},
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\install_lib.py", line 11, in run
    self.build()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\command\install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\build_ext.py", line 84, in run
    _build_ext.run(self)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\Cython\Distutils\old_build_ext.py", line 186, in run
    _build_ext.build_ext.run(self)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\utils\cpp_extension.py", line 580, in build_extensions
    build_ext.build_extensions(self)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\Cython\Distutils\old_build_ext.py", line 195, in build_extensions
    _build_ext.build_ext.build_extensions(self)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self._build_extensions_serial()
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\command\build_ext.py", line 474, in _build_extensions_serial
    self.build_extension(ext)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\setuptools\command\build_ext.py", line 205, in build_extension
    _build_ext.build_extension(self, ext)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\distutils\command\build_ext.py", line 534, in build_extension
    depends=ext.depends)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\utils\cpp_extension.py", line 562, in win_wrap_ninja_compile
    with_cuda=with_cuda)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\utils\cpp_extension.py", line 1140, in _write_ninja_file_and_compile_objects
    error_prefix='Error compiling objects for extension')
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\utils\cpp_extension.py", line 1413, in _run_ninja_build
    raise RuntimeError(message)
RuntimeError: Error compiling objects for extension
```

**Comment by anvuongb** [comment]
> 
> 
> ```
> DCNv2/src/cuda/dcn_v2_cuda.cu(107): error: identifier "THCState_getCurrentStream" is undefined
> ```
> 
> Any updates on this issue?

I'm also having this problem, have you found any solutions?

**Comment by HouGall** [comment]
> Error compiling objects for extension

hello,i got the same problem ,so have you found any solutions? 
pytorch1.5+cuda10.1,i got RuntimeError: Error compiling objects for extension

**Comment by zhangkui669** [comment]
> ound any solutions?

+1, cuda10.2, torch1.5

**Comment by zhangkui669** [comment]
> > ```
> > DCNv2/src/cuda/dcn_v2_cuda.cu(107): error: identifier "THCState_getCurrentStream" is undefined
> > ```
> > 
> > 
> > Any updates on this issue?
> 
> I'm also having this problem, have you found any solutions?

+1, torch1.5, cuda10.2

**Comment by trobro** [comment]
Sorry guys this issue only concerned this error:

    unresolved external symbol "struct THCState * state"

And that was fixed in #46 so I'm closing this. Please create new issues for the other errors you see when building.
